client/Components/Pages/Signup.jsx

```diff
@@ -39,30 +42,43 @@ class Signup extends Component {
     });
   }
 
-  handleSubmit(e) {
-    e.preventDefault();
+  handleSubmit(event) {
+    event.preventDefault();
+    if(this.state.username.trim() === '' 
```

Comment:
Cut down the values here by assigning them to variables.

Reply:
Ok, will do that, thanks.